api/source/service/mysql/STIGService.js

```diff
@@ -969,7 +1054,39 @@ exports.getCci = async function(cci, userObject) {
  **/
 exports.getCcisByRevision = async function(benchmarkId, revisionStr, userObject) {
   try {
-    let rows = await _this.METHOD()
+
+    let [input, version, release] = /V(\d+)R(\d+(\.\d+)?)/.exec(revisionStr)
```

Comment:
Unfortunately this approach, which is incorrectly present elsewhere in this file, is insufficient. It throws if revisionStr == "latest", which is a legal value per the the OAS.

The various service methods handle "latest" in various ways. In this case, please consider using the approach in STIGService.js line 79. 
```
  if (inPredicates.revisionStr != 'latest') {
    joins = ['revision r']
    let [results, version, release] = /V(\d+)R(\d+(\.\d+)?)/.exec(inPredicates.revisionStr)
    predicates.statements.push('r.version = ?')
    predicates.binds.push(version)
    predicates.statements.push('r.release = ?')
    predicates.binds.push(release)
  } else {
    joins = ['current_rev r']
  }
```
This approach does require this query be decomposed into arrays for columns, joins, predicates. This must be the only complex query in the API that is not constructed using that pattern

Reply:
@cd-rite Giving you a heads up that the newman test does not make a request to this endpoint using "latest", so this code is passing when it should not.

Reply:
reworked to add handling of latest